nav2_simple_commander/nav2_simple_commander/footprint_collision_checker.py

```diff
@@ -0,0 +1,96 @@
+#! /usr/bin/env python3
+# Copyright 2021 Samsung Research America
+# Copyright 2022 Afif Swaidan
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from math import cos, sin
+from line_iterator import LineIterator
+from costmap_2d import PyCostmap2D
+from geometry_msgs.msg import Point
```

Comment:
Instead, take in a geometry_msgs/Polygon (which has that as one of its fields essentially)

Reply:
Will do that, thanks for the notice!